### ISSUE TITLE
littleb.c:1579:5: buffer owerflow

### ISSUE THREAD

**Issue by plusky** [intel-iot-devkit/littleb#1]
When bulding lillteb with following flags `-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables` 

following warning appears:

```
[    3s] /home/abuild/rpmbuild/BUILD/littleb-0.1.0/src/littleb.c:1466:12: warning: unused variable 'i' [-Wunused-variable]
[    3s]      int r, i;
[    3s]             ^
[    3s] In file included from /usr/include/stdio.h:936:0,
[    3s]                  from /home/abuild/rpmbuild/BUILD/littleb-0.1.0/api/littleb.h:29,
[    3s]                  from /home/abuild/rpmbuild/BUILD/littleb-0.1.0/src/littleb.c:24:
[    3s] In function 'snprintf',
[    3s]     inlined from 'lb_register_characteristic_read_event' at /home/abuild/rpmbuild/BUILD/littleb-0.1.0/src/littleb.c:1579:5:
[    3s] /usr/include/bits/stdio2.h:64:10: warning: call to __builtin___snprintf_chk will always overflow destination buffer
[    3s]    return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
[    3s]           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[    3s]         __bos (__s), __fmt, __va_arg_pack ());
[    3s]         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Strongly suggesting that there is potential buffer overflow

**Comment by Hbrinj** [comment]
Hi, thanks I'll take a look :)

**Comment by Hbrinj** [comment]
fixed in commit 99f459348fb2b5b067ba098478eef284c0d2516f thanks!
